### DISCUSSION TITLE
[10 gennaio 2024] Calcolabilità

### DISCUSSION THREAD

**Discussion by matypist** [sapienzastudentsnetwork/automi-calcolabilita-complessita#20] [Calcolabilità]
![image](https://github.com/sapienzastudentsnetwork/acc2324/assets/69389461/5f6b480d-2913-4359-91c8-d4f4453d995d)

**Comment by Exyss** [comment]
Per il primo punto, ripropongo la soluzione presente sui miei appunti:

![image](https://github.com/sapienzastudentsnetwork/acc2324/assets/65043207/1cace7b4-04ef-41bf-b93a-04fe15e789d9)

----

La dimostrazione richiesta nel secondo punto è presente sul libro di testo

**Comment by aflaag** [comment]
Oltre alla dimostrazione fornita da @Exyss , aggiungo che è possibile dimostrare l'indecidibilità di $REJ_\textsf{TM}$ utilizzando la stessa dimostrazione dell'indecidibilità di $A_\textsf{TM}$, come segue:
- per assurdo, sia $REJ_\textsf{TM} \in \textsf{DEC}$, e dunque esiste un decisore $H$ di $REJ_\textsf{TM}$, che su input $\langle M, w \rangle$, accetta se e solo se $M$ rifiuta $w$;
- allora, sia $D$ il seguente decisore, che chiama $H$ quando l'input di $M$ è la descrizione di sé stessa: $$D(\langle M \rangle ) \ \textrm{accetta} \iff H(\langle M, \langle M \rangle \rangle) \ \textrm{accetta}$$
- di conseguenza, su input $\langle D \rangle$ stesso, $D$ computa come segue: $$D(\langle D \rangle) \ \textrm{accetta} \iff H(\langle D, \langle D \rangle \rangle) \ \textrm{accetta} \iff D \ \textrm{rifiuta} \ \langle D \rangle$$ il che è chiaramente assurdo.

**Reply by DomoticLab** [reply]
Salve, se lo scrivo in notazione formale l'editor lo pubblica omettendo alcuni valori. Pertanto lo scrivo in questo modo: D su input M non inverte il risultato fornito da H su input M M ?
